cardano-testnet/src/Testnet/Start/Cardano.hs

```diff
@@ -198,6 +197,62 @@ cardanoTestnet
 
   _ <- createSPOGenesisAndFiles testnetOptions genesisOptions mShelleyGenesis mAlonzoGenesis mConwayGenesis (TmpAbsolutePath tmpAbsPath)
 
+  nodeConfigFile <- case cardanoNodes of
+    AutomaticNodeOptions _ -> do
+      configurationFile <- H.noteShow $ tmpAbsPath </> "configuration.yaml"
+      -- Add Byron, Shelley and Alonzo genesis hashes to node configuration
+      config <- createConfigJson (TmpAbsolutePath tmpAbsPath) sbe
+      H.evalIO $ LBS.writeFile configurationFile config
+      return configurationFile
+    UserProvidedNodeOptions userSubmittedNodeConfigFile ->
+      liftIO $ IO.makeAbsolute userSubmittedNodeConfigFile
+
+  case configFilesBehaviour of
+    OnlyGenerate ->
```

Comment:
Should be abstracted out as a separate function otherwise it just adds to the already cluttered `cardanoTestnet`. We're not starting a testnet in the case of configuration generation so it makes sense for this to exist as a standalone function.